src/graderPublic/java/h09/sequence/operation/FlatteningTransformingSequenceBasicTest.java

```diff
@@ -50,8 +50,8 @@ void testIteratorBasic() {
         Sequence<Integer> limitedSeq = new LimitSequence<>((Sequence<Integer>) seq, 6);
         Iterator<Integer> it = limitedSeq.iterator();
         for (int i = 0; i < 6; i++) {
-            Assertions.assertTrue(it.hasNext(), "FlatteningTransformingSequence should have 6 elements");
-            Assertions.assertEquals(i + 1, it.next(), "FlatteningTransformingSequence should have elements 1, 2, 3, 4, 5, 6");
+            Assertions.assertTrue(it.hasNext(), i == 0 ? "Resulting sequence should have 6 elements! Has: 1 element" : "Resulting sequence should have 6 elements! Has: " + (i + 1) + " elements");
```

Comment:
Off-by-one error - if it fails at i == 0 then there are 0 elements not 1

Reply:
Should be fixed now